### ISSUE TITLE
device-readings does not seem to have any content

### ISSUE THREAD

**Issue by ralpje** [printix/Power-BI#9]
Hi,

When using the script for partners, on one of the connected tenants the script reports a warning 'device-readings does not seem to have any content'. This is a logged as a warning, and not a breaking error. 

However, the data import in PowerBI seems to fail on this quirk.

```
Something went wrong

There was an error when processing the data in the dataset.
Please try again later or contact support. If you contact support, please provide these details.
MessageAzureBlobs failed to get contents from 'https://blahblah.blob.core.windows.net/customername/device-readings/device-readings.json'. Status code: 404, description: 'The specified blob does not exist.'. Table: Device_readings.
```
Of course, this is to be expected because the blob isn't created by the script.
How can I correct this to make sure the data import in PowerBI works correctly, even when some data in the dataset is missing?

**Comment by MortenRei** [comment]
Hi @ralpje 

This is most likely caused by one of two things;
- There is no devices in their printix tenant
- There has been zero prints in the time frame you specified.

It's correct that the PowerBI dashboard will fail. This is actually by design. The reason for that is because if you had a successfull extract, the device-readings file will be there, with content. So a failure in the future will not break anything, it's just for the first extract.

**Comment by ralpje** [comment]
Hi @MortenRei,

Well, there actually is at least one active printer in the tenant and the printix portal shows 579 pages printed. 
Is there any way I can check any deeper what data is coming in through the script?

**Comment by ralpje** [comment]
Hi @MortenRei, 

Apologies for the late reply. I've checked the runbook today, but all of a sudden it seems to be working. I think the issue was triggered by the fact that I 'recreated' some printers / print queues, which caused the printer to not be used for printing when the script ran. All prints that were previously done were registered to a different printer, that was not existent (because I deleted it) at the moment the script ran. 

Anyway, the issue seems to be resolved and my PowerBI report is updated correctly. Thanks for your assistance.
